var/spack/repos/builtin/packages/py-shiboken/package.py

```diff
@@ -18,13 +18,21 @@ class PyShiboken(PythonPackage):
 
     depends_on("cmake", type="build")
 
+    # to prevent error: 'PyTypeObject' {aka 'struct _typeobject'} has no member
+    # named 'tp_print'
+    depends_on("python@:3.8", type=("build", "run"))
     depends_on("py-setuptools", type="build")
     # in newer pip versions --install-option does not exist
     depends_on("py-pip@:23.0", type="build")
     depends_on("py-sphinx@:3.4", type=("build", "run"))
     depends_on("libxml2")
     depends_on("qt@:4.8")
 
+    # Fails to query the Qt version with qmake with qt@3
+    conflicts("^qt@3")
+    # to prevent ImportError: cannot import name 'soft_unicode' from 'markupsafe'
+    conflicts("^py-markupsafe@2.0.2:")
```

Comment:
Package doesn't use markupsafe, I think this belongs somewhere else.

Reply:
As far as I can tell, `py-markupsafe` comes in as dependency of `py-jinja` which is a dependency of `py-sphinx`, and

concretization without `py-markupsafe` conflict:
```
[+]      ^py-sphinx@3.4.1%gcc@12.3.1 build_system=python_pip arch=linux-fedora37-x86_64
...
[+]          ^py-jinja2@2.11.3%gcc@12.3.1~i18n build_system=python_pip arch=linux-fedora37-x86_64
[+]              ^py-markupsafe@2.1.3%gcc@12.3.1 build_system=python_pip arch=linux-fedora37-x86_64
...
```

concretization with `py-markupsafe` conflict:
```
[+]      ^py-sphinx@3.4.1%gcc@12.3.1 build_system=python_pip arch=linux-fedora37-x86_64
...
[+]          ^py-jinja2@2.11.3%gcc@12.3.1~i18n build_system=python_pip arch=linux-fedora37-x86_64
[+]              ^py-markupsafe@2.0.1%gcc@12.3.1 build_system=python_pip arch=linux-fedora37-x86_64
...
```

Reply:
I have removed the `py-markupsafe` conflict here as well. Once #38913 is merged this issue is solved.